src/kernel/arch/x86/isr.zig

```diff
@@ -212,4 +302,182 @@ pub fn init() void {
     openIsr(30, isr30);
     openIsr(31, isr31);
     openIsr(syscalls.INTERRUPT, isr128);
+
+    log.logInfo("Done\n");
+
+    if (build_options.rt_test) runtimeTests();
+}
+
+extern fn testFunction0() void {}
+fn testFunction1(ctx: *arch.InterruptContext) void {}
+fn testFunction2(ctx: *arch.InterruptContext) void {}
+fn testFunction3(ctx: *arch.InterruptContext) void {}
+fn testFunction4(ctx: *arch.InterruptContext) void {}
+
+test "openIsr" {
+    idt.initTest();
+    defer idt.freeTest();
+
+    const index = u8(0);
+    const handler = testFunction0;
+    const ret: idt.IdtError!void = {};
+
+    idt.addTestParams("openInterruptGate", index, handler, ret);
+
+    openIsr(index, handler);
+}
+
+test "isValidIsr" {
+    comptime var i = 0;
+    inline while (i < NUMBER_OF_ENTRIES) : (i += 1) {
+        expectEqual(true, isValidIsr(i));
+    }
+
+    expectEqual(true, isValidIsr(syscalls.INTERRUPT));
+
+    expectEqual(false, isValidIsr(200));
+}
+
+test "registerIsr re-register syscall handler" {
+    // Pre testing
+    if (syscall_handler) |_| {
+        std.debug.panic("Pre-testing failed");
+    }
+
+    // Call function
+    try registerIsr(syscalls.INTERRUPT, testFunction3);
+    expectError(IsrError.IsrExists, registerIsr(syscalls.INTERRUPT, testFunction4));
+
+    // Post testing
+    if (syscall_handler) |h| {
+        // Should still have the original handler
+        expectEqual(testFunction3, h);
+    } else {
+        std.debug.panic("Pre-testing failed");
```

Comment:
Why `std.debug.panic` and not `@panic'?

Reply:
have removed the panics and use `expect`'s